pkg/tcpip/network/ipv4/ipv4.go

```diff
@@ -357,7 +357,7 @@ func (e *endpoint) HandlePacket(r *stack.Route, pkt tcpip.PacketBuffer) {
 		}
 		var ready bool
 		var err error
-		pkt.Data, ready, err = e.fragmentation.Process(hash.IPv4FragmentHash(h), h.FragmentOffset(), last, more, pkt.Data)
+		pkt.Data, ready, err = e.fragmentation.Process(hash.IPv4FragmentHash(h), h.FragmentOffset(), last, more, pkt.Data, r, headerView)
```

Comment:
you can just pass pkt to this one as it already includes the netHeader.

Reply:
we have taken care of this comment.